fast_llm/functional/cross_entropy.py

```diff
@@ -145,13 +145,19 @@ def _fused_cross_entropy_forward_backward(
     else:
         predicted_logits = (target * logits_norm).sum(dim=-1, keepdim=True)
 
+    # shouldn't the predicted_logits be scaled by the number of ranks so that the average loss is correct? i.e.
```

Comment:
Not sure what you mean here. We sum over the contributions from each TP rank, so the result is the total predicted logits, and scaling it doesn't make sense.